StackoverflowChatbot/Discord.cs

```diff
@@ -88,6 +89,28 @@ private static Task ClientRecieved(SocketMessage arg)
 			return Task.CompletedTask;
 		}
 
+        private static string BuildSoMessage(SocketGuildUser user, Config.Base config, SocketMessage arg)
+        {
+			var displayname = string.IsNullOrEmpty(user.Nickname) ? user.Username : user.Nickname;
+            var messageStart = $@"\[**[{displayname}]({config.DiscordInviteLink})**]";
+			var messageContent = arg.Content;
+			foreach (var mentionedUser in arg.MentionedUsers)
+			{
+				messageContent.Replace(mentionedUser.Mention, $"@{mentionedUser.Username}");
+			}
+			foreach (var mentionedRoles in arg.MentionedRoles)
+			{
+				messageContent.Replace(mentionedRoles.Mention, $"[@{mentionedRoles.Name}]({config.DiscordInviteLink})");
+			}
+			
+			var embeddedCode = Regex.Matches(messageContent, "```.+```", RegexOptions.Multiline);
+			foreach (Match codeBlock in embeddedCode)
+			{
+				var soCodeBlock = codeBlock.ToString().Replace("\n", "\n    ");
```

Comment:
This needs to be spaces at the start of every line, not just at the start of the whole block

Reply:
it replaces every newline with a newline and 4 spaces, which in theory should do exactly that

Reply:
Oh right yeah I didn't read it properly haha